llvm/lib/Transforms/Utils/LowerMemIntrinsics.cpp

```diff
@@ -456,6 +456,56 @@ static void createMemMoveLoop(Instruction *InsertBefore, Value *SrcAddr,
   ElseTerm->eraseFromParent();
 }
 
+static void createMemSetPatternLoop(Instruction *InsertBefore, Value *DstAddr,
+                                    Value *Count, Value *SetValue,
+                                    Align DstAlign, bool IsVolatile) {
+  BasicBlock *OrigBB = InsertBefore->getParent();
+  Function *F = OrigBB->getParent();
+  const DataLayout &DL = F->getDataLayout();
+
+  if (DL.isBigEndian())
+    report_fatal_error("memset.pattern expansion not currently "
+                       "implemented for big-endian targets",
+                       false);
+
+  if (!isPowerOf2_32(SetValue->getType()->getScalarSizeInBits()))
+    report_fatal_error("Pattern width for memset_pattern must be a power of 2",
+                       false);
```

Comment:
This limitation would have to be checked by the IR verifier. I don't think it's needed though, your code should work fine without it.

Reply:
I've removed this limitation and added test coverage.